### PR TITLE
kv-client: use worker pool in region worker

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Part of #1393 

To increase the throughput of kv client, we should use multiple goroutines in region worker

### What is changed and how it works?

- Reuse `github.com/pingcap/ticdc/pkg/workerpool`

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @liuzix @leoppro @overvenus

File: cdc/kv/client_test.go [hunk 1]
```diff
@@ -45,7 +46,15 @@ import (
 	"google.golang.org/grpc"
 )
 
-func Test(t *testing.T) { check.TestingT(t) }
+func Test(t *testing.T) {
+	conf := config.GetDefaultServerConfig()
+	config.StoreGlobalServerConfig(conf)
```
Comment (leoppro) [inline cdc/kv/client_test.go:51]: should we set a default config in `init` function?

File: cdc/server.go [hunk 2]
```diff
@@ -263,6 +264,10 @@ func (s *Server) run(ctx context.Context) (err error) {
 		return sorter.RunWorkerPool(cctx)
 	})
 
+	wg.Go(func() error {
```
Comment (leoppro) [inline cdc/server.go:267]: maybe we can run worker pool in `init` function?

**Review by leoppro — APPROVED** [review]

File: cdc/server.go (hunk 2 above)
Comment (amyangfei) [reply cdc/server.go:267]: yep both run it in `init` function and explicit call work, but imho the `init` function is a little dangerous in some scenarios, which may make initialization step unclear (such as init sequence, unexpected init when we import a package but don't want it to init, etc)
such as the discussion in this article: https://medium.com/higher-order-functions/golang-how-to-sensibly-use-init-function-8077a4ec15e5

File: pkg/config/config.go [hunk 3]
```diff
@@ -161,6 +161,10 @@ var defaultServerConfig = &ServerConfig{
 		SortDir:                "/tmp/cdc_sort",
 	},
 	Security: &SecurityConfig{},
+	KVClient: &KVClientConfig{
+		WorkerConcurrent: 8,
+		WorkerPoolSize:   0, // 0 will use NumCPU() * 2
```
Comment (overvenus) [inline pkg/config/config.go:166]: Do we need an upper limit of pool size?

File: cdc/kv/region_worker.go [hunk 4]
```diff
@@ -617,3 +785,35 @@ func (w *regionWorker) evictAllRegions(ctx context.Context) error {
...
+		return
+	}
+	workerPoolOnce.Do(func() {
+		cfg := config.GetGlobalServerConfig().KVClient
+		var size int
+		if cfg.WorkerPoolSize > 0 {
+			size = cfg.WorkerPoolSize
+		} else {
+			size = runtime.NumCPU() * 2
+		}
```
Comment (overvenus) [inline cdc/kv/region_worker.go:802]: Is there any test that covers these lines?

Comment (amyangfei) [reply cdc/kv/region_worker.go]: added unit test

File: pkg/config/config.go (hunk 3 above)
Comment (amyangfei) [reply pkg/config/config.go:167]: add an upper limit 64

**Comment by amyangfei** [comment]
ptal @overvenus @liuzix

File: cdc/kv/region_worker.go [hunk 5]
```diff
@@ -15,28 +15,43 @@ package kv
...
 	"go.uber.org/zap"
 	"golang.org/x/sync/errgroup"
 	"golang.org/x/time/rate"
 )
 
+var (
+	regionWorkerPool workerpool.WorkerPool
+	workerPoolOnce   sync.Once
+	// TODO: unified channel buffer mechanism
+	regionWorkerInputChanSize = 128000
```
Comment (JinLingChristopher) [inline cdc/kv/region_worker.go:45]: can you add a comment about this magic 128000?

Comment (amyangfei) [reply cdc/kv/region_worker.go:48]: added some comments

**Pull request merged**